sdk-core/src/main/java/care/data4life/sdk/RecordService.kt

```diff
@@ -913,13 +908,9 @@ class RecordService internal constructor(
         val resource = record.resource
 
         if (!fhirAttachmentHelper.hasAttachment(resource)) return record
-        val attachments = fhirAttachmentHelper.getAttachment(resource) as List<Any?>?
+        val attachments = fhirAttachmentHelper.getAttachment(resource)
                 ?: return record
 
-        if (record.attachmentsKey == null) {
```

Comment:
Why is this removed? Isn't there a case the ther is no attachment key yet?

Reply:
It is actually not removed, it is just moved (as mentioned in the PR before) into `resolveAttachmentKey`, which is called by `uploadAttachmentsOnDemand `.

Reply:
👍